cmd/influxd/inspect/dump_tsi/dump_tsi.go

```diff
@@ -0,0 +1,486 @@
+	showTagValues      bool
+	showTagValueSeries bool
+
+	measurementFilter *regexp.Regexp
+	tagKeyFilter      *regexp.Regexp
+	tagValueFilter    *regexp.Regexp
+
+	w io.Writer
+}
+
+func NewDumpTSICommand() *cobra.Command {
+	var arguments args
+	var measurementFilter, tagKeyFilter, tagValueFilter string
+	cmd := &cobra.Command{
+		Use:   "dumptsi",
+		Short: "Dumps low-level details about tsi1 files.",
+		Args:  cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Parse filters.
+			if measurementFilter != "" {
+				re, err := regexp.Compile(measurementFilter)
+				if err != nil {
+					return fmt.Errorf("failed to parse regex %q: %w", measurementFilter, err)
+				}
+				arguments.measurementFilter = re
+			}
+			if tagKeyFilter != "" {
+				re, err := regexp.Compile(tagKeyFilter)
+				if err != nil {
+					return fmt.Errorf("failed to parse regex %q: %w", tagKeyFilter, err)
+				}
+				arguments.tagKeyFilter = re
+			}
+			if tagValueFilter != "" {
+				re, err := regexp.Compile(tagValueFilter)
+				if err != nil {
+					return fmt.Errorf("failed to parse regex %q: %w", tagValueFilter, err)
+				}
+				arguments.tagValueFilter = re
+			}
+
+			arguments.paths = args
+			if len(arguments.paths) == 0 {
+				return fmt.Errorf("at least one path required")
+			}
+
+			// Some flags imply other flags.
+			if arguments.showTagValueSeries {
+				arguments.showTagValues = true
+			}
+			if arguments.showTagValues {
+				arguments.showTagKeys = true
+			}
+			if arguments.showTagKeys {
+				arguments.showMeasurements = true
+			}
+
+			arguments.w = cmd.OutOrStdout()
+			return arguments.run()
+		},
+	}
+
+	cmd.Flags().StringVar(&arguments.seriesFilePath, "series-file", "",
+		"Path to series file")
+	cmd.Flags().BoolVar(&arguments.showSeries, "series", false,
+		"Show raw series data")
+	cmd.Flags().BoolVar(&arguments.showMeasurements, "measurements", false,
+		"Show raw measurement data")
+	cmd.Flags().BoolVar(&arguments.showTagKeys, "tag-keys", false,
+		"Show raw tag key data")
+	cmd.Flags().BoolVar(&arguments.showTagValues, "tag-values", false,
+		"Show raw tag value data")
+	cmd.Flags().BoolVar(&arguments.showTagValueSeries, "tag-value-series", false,
+		"Show raw series data for each value")
+	cmd.Flags().StringVar(&measurementFilter, "measurement-filter", "",
+		"Regex measurement filter")
+	cmd.Flags().StringVar(&tagKeyFilter, "tag-key-filter", "",
+		"Regex tag key filter")
+	cmd.Flags().StringVar(&tagValueFilter, "tag-value-filter", "",
+		"Regex tag value filter")
+
+	cmd.MarkFlagRequired("series-file")
+
+	return cmd
+}
+
+func (a *args) run() error {
+	sfile := tsdb.NewSeriesFile(a.seriesFilePath)
+	if err := sfile.Open(); err != nil {
+		return err
+	}
+	defer sfile.Close()
+
+	// Build a file set from the paths on the command line.
+	idx, fs, err := a.readFileSet(sfile)
+	if err != nil {
+		return err
+	}
+	if fs != nil {
+		defer fs.Release()
+		defer fs.Close()
+	}
+	defer idx.Close()
```

Comment:
I think this needs to be wrapped in an `if idx != nil`, it looks like it'll be `nil` if `fs != nil`